File: Lista_Logica_Programacao/Java/Ex4.java
// Exercício 4 - Tabuada de um número

import java.util.Scanner;

public class Ex4 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int numero = sc.nextInt();

        for (int i = 1; i <= 10; i++) {
            System.out.println(numero + " x " + i + " = " + (numero * i));
        }

        sc.close();
    }
}
/*
Anotações:

- Usei for tradicional porque Java precisa declarar a variável do contador
- int i = 1 começa no 1
- i <= 10 faz repetir até 10
- i++ incrementa de 1 em 1
- Precisei colocar (numero * i) entre parênteses porque senão ele poderia concatenar errado com string
*/

File: Lista_Logica_Programacao/Python/ex4.py
# Exercício 4 - Tabuada

numero = int(input("Digite um número: "))

# usei for para repetir de 1 até 10
for i in range(1, 11):
    print(numero, "x", i, "=", numero * i)

# Anotações:
# - Usei for porque preciso repetir 10 vezes
# - range(1, 11) vai do 1 até o 10 (o 11 não entra)
# - numero * i faz a multiplicação da tabuada
# - Python é simples porque o for já funciona direto com range